### ISSUE TITLE
Not finding requirements-file for Automattic/action-required-review

### ISSUE THREAD

**Issue by charliegroll** [Automattic/jetpack#22186]
### Impacted plugin

None / Other

### Steps to Reproduce

Note that this is an initial PR and `required-review.yml` is a brand new file

.github/workflows/required-review.yml
```yaml
set_reviewers:
      runs-on: ubuntu-latest
      steps:
      - uses: Automattic/action-required-review@v2
        with:
          requirements-file: .github/required-review.yml
```

.github/required-review.yml
```yaml
- name: Security
  paths: unmatched
  teams:
    - @charliegroll
```

I tried to do this inline with `requirements` but it complains about the indentation when it gets to the evaluation of the yaml string 🙃 

When I try inline with `@charliegroll` I get 

```yaml
- uses: Automattic/action-required-review@v2
        with:
          requirements: |
            - paths: unmatched
              teams:
                - @charliegroll
```

```log
Run Automattic/action-required-review@v2
Error: bad indentation of a sequence entry (3:7)

 1 | - paths: unmatched
 2 |   teams:
 3 |     - @charliegroll
-----------^
YAMLException: bad indentation of a sequence entry (3:7)

 1 | - paths: unmatched
 2 |   teams:
 3 |     - @charliegroll
-----------^
    at generateError (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:183:1)
    at throwError (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:187:1)
    at readBlockSequence (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1015:1)
    at composeNode (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1440:1)
    at readBlockMapping (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1164:1)
    at composeNode (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1441:1)
    at readBlockSequence (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1008:1)
    at composeNode (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1440:1)
    at readDocument (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1625:1)
    at loadDocuments (/home/runner/work/_actions/Automattic/action-required-review/node_modules/.pnpm/js-yaml@4.1.0/node_modules/js-yaml/lib/loader.js:1688:1)
```

but I know that it works with non-`@` teams because I get the following for just `security` (which doesn't exist):

```log
Run Automattic/action-required-review@v2
Loaded 1 review requirement(s)
Found 0 reviewer(s)
PR affects 3 file(s)
Checking requirement "#0"...
  Matches the following files:
     - .github/required-review.yml
     - .github/workflows/require-reviewers-for-changes.yml
     - test.js
  Checking reviewers...
    Union of these:
  Error: Failed to query netlify team security from GitHub
  WrappedError: Failed to query netlify team security from GitHub
      at fetchTeamMembers (/home/runner/work/_actions/Automattic/action-required-review/v2/webpack:/required-review/src/team-members.js:37:1)
      at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

### A clear and concise description of what you expected to happen.

I expect it to find my requirements file.

### What actually happened

```log
Run Automattic/action-required-review@v2
  with:
    requirements-file: .github/required-review.yml
    status: Required review
    token: ***
Error: ENOENT: no such file or directory, open '.github/required-review.yml'
Error: ENOENT: no such file or directory, open '.github/required-review.yml'
    at Object.openSync (fs.js:440:3)
    at Object.readFileSync (fs.js:342:35)
    at getRequirements (/home/runner/work/_actions/Automattic/action-required-review/v2/webpack:/required-review/src/main.js:27:1)
    at main (/home/runner/work/_actions/Automattic/action-required-review/v2/webpack:/required-review/src/main.js:59:1)
    at /home/runner/work/_actions/Automattic/action-required-review/v2/webpack:/required-review/src/main.js:116:1
    at /home/runner/work/_actions/Automattic/action-required-review/v2/dist/index.js:16766:3
    at Object.<anonymous> (/home/runner/work/_actions/Automattic/action-required-review/v2/dist/index.js:16769:12)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
```

### Other information

Using https://github.com/Automattic/jetpack/blob/master/.github/workflows/required-review.yml as a reference, I'm not sure what the issue could be

### Browser

Firefox

**Comment by charliegroll** [comment]
P.S. sorry for the compound issues - I can open a separate issue for the inline/indentation issue, but I really just want to get either one working for a start

**Comment by jeherve** [comment]
> I tried to do this inline with requirements but it complains about the indentation when it gets to the evaluation of the yaml string 🙃

Could you try this?

```yml
- paths:
  - 'unmatched'
  teams:
    - '@charliegroll'
```

****************

> `Error: ENOENT: no such file or directory, open '.github/required-review.yml'`

Could it be because this is in a PR, using the `pull_request_review` event, thus looking at your repository's main branch where the file does not exist yet? What happens when you first commit that file, and then add the workflow?

**Comment by charliegroll** [comment]
@jeherve I think the first part of that resolved my issue (the quotes) 

```log
Checking requirement "#0"...
  Matches the following files:
     - .github/workflows/require-reviewers-for-changes.yml
     - test.js
  Checking reviewers...
    Union of these:
      Members of @charliegroll: <empty set>
    => <empty set>
Error: Requirement "#0" is not satisfied by the existing reviews.
```

but the action is stuck in `Pending` - is that expected, or am I meant to detect this and fail the workflow myself?

**Comment by jeherve** [comment]
> the action is stuck in Pending

It should remain as "Pending" until a review approval is left. Could you give that a try?

**Comment by charliegroll** [comment]
That works - I don't think that's desired since we still want to know when our builds pass/fail and I'll want to surface whose review is required, if any, but I'm unblocked so thanks for the help @jeherve

**Comment by jeherve** [comment]
> I don't think that's desired since we still want to know when our builds pass/fail and I'll want to surface whose review is required

That makes sense. What would you have the test be like until the review in this case?

**Comment by charliegroll** [comment]
My personal preference is to fail the build - ours take several minutes (until recently, one repo was taking 11-15 minutes and at a previous company it was like 40 minutes... 💀 ) so my current plan is to detect that a reviewer is needed, post to the PR as a bot and tag the necessary reviewer(s), and fail the build. my action is going to reevaluate on every commit and review, so I'd like `pending` to really mean `running` and not just "waiting on checks"

**Comment by jeherve** [comment]
> My personal preference is to fail the build

Wouldn't that be confusing for some? A failing build usually means something is wrong; a PR author may see the failing build and think they need to do something to fix things, when in fact what's missing is an approval from someone else.

**Comment by jeherve** [comment]
> What's the intended feedback mechanism to let PR authors (and required reviewers, for that matter) know that a review from a specific team/individual is required? 

So far, the non-completed check has been the mechanism used to let PR authors they needed an approval.

> in my scenario, you immediately see that it's failed and check for which action is failing, which I envision points you to whose review is required and why.

That's a good point, I can see how this could be useful.

The issue with a failing test as the default result (until someone approves the PR) is that you then do not have a quick way to know if the check if failing because of an issue with the workflow, or just because no one approved the PR yet.

**Comment by anomiex** [comment]
> ```yaml
> set_reviewers:
>       runs-on: ubuntu-latest
>       steps:
>       - uses: Automattic/action-required-review@v2
>         with:
>           requirements-file: .github/required-review.yml
> ```

The problem there is that you didn't check out your repo. Try this:
```yaml
set_reviewers:
      runs-on: ubuntu-latest
      steps:
      - uses: actions/checkout@v2
      - uses: Automattic/action-required-review@v2
        with:
          requirements-file: .github/required-review.yml
```

**Comment by anomiex** [comment]
As for the pending versus failed question, I agree with @jeherve that it makes more sense for it to be pending when the state is "awaiting reviews". But if someone were to submit a PR adding a "fail-if-reviews-required" option, I'd review it.

**Comment by anomiex** [comment]
Closing this now. The titular "not finding requirements file" issue was because the repo wasn't being checked out first, and the "fail instead of pending" feature request was done in #25091.
